novawallet/Common/Services/TransactionSubscription/ExtrinsicProcessor+Fee.swift

```diff
@@ -35,25 +40,85 @@ extension ExtrinsicProcessor {
         }
     }
 
+    func findAssetsFee(
+        for index: UInt32,
+        sender: AccountId,
+        eventRecords: [EventRecord],
+        codingFactory: RuntimeCoderFactoryProtocol
+    ) -> Fee? {
+        findAssetsCustomFee(
+            for: index,
+            eventRecords: eventRecords,
+            codingFactory: codingFactory
+        )
+            ?? findFee(
+                for: index,
+                sender: sender,
+                eventRecords: eventRecords,
+                metadata: codingFactory.metadata,
+                runtimeJsonContext: codingFactory.createRuntimeJsonContext()
+            )
+    }
+
+    func findOrmlFee(
+        for params: HydraSwapExtrinsicParsingParams,
+        extrinsicIndex: UInt32,
+        eventRecords: [EventRecord],
+        codingFactory: RuntimeCoderFactoryProtocol
+    ) throws -> Fee? {
+        if
+            let customFee = try findHydraCustomFee(
+                in: eventRecords,
+                swapEvents: params.events,
+                codingFactory: codingFactory
+            ) {
+            return customFee
+        } else {
+            let optNativeFee = findFee(
+                for: extrinsicIndex,
+                sender: params.sender,
+                eventRecords: eventRecords,
+                metadata: codingFactory.metadata,
+                runtimeJsonContext: codingFactory.createRuntimeJsonContext()
+            )
+
+            guard
+                let feeAssetId = chain.utilityChainAssetId()?.assetId,
+                let nativeFee = optNativeFee?.amount
+            else {
+                return nil
+            }
+
+            return Fee(
+                amount: nativeFee,
+                assetId: feeAssetId
```

Comment:
In other place, I saw we returning nil for native fee

Reply:
Oh, I will check before the merge. Thanks.